src/components/contentAvail/contentAvailSrv.js

```diff
@@ -1,160 +1,230 @@
+                    return !!(purchaseData.exam === PURCHASED_ALL  || purchaseData.exam[examKeyProp]);
+                }
 
-                    return _isFreeContent(freeContent,['exam',idToKeyInStorage(examId)]);
-                });
-            };
+                function _isFreeContent(freeContentData,pathArr){
+                    var fullPath = pathArr.join('.');
+                    var isFreeGetter = $parse(fullPath);
+                    return !!isFreeGetter(freeContentData);
+                }
+
+                function hasSubscription() {
+                    return getUserPurchaseData().then(function(purchaseData){
+                        return _hasSubscription(purchaseData.subscription);
+                    });
+                }
 
-            this.isSectionAvail = function(examId,sectionId){
-                return _baseIsEntityAvail().then(function(res){
-                    if(res === true){
-                        return true;
+                function isDailyAvail(dailyOrder){
+                    if(!angular.isNumber(dailyOrder) || isNaN(dailyOrder)){
+                        return $q.reject('daily order should be a number');
                     }
+                    return _baseIsEntityAvail().then(function(res){
+                        if(res === true){
+                            return true;
+                        }
+
+                        var purchaseData = res[0];
+                        var freeContent = res[1];
+                        var specials = res[2];
+
+                        if((specials.daily + freeContent.daily) >= dailyOrder){
+                            return true;
+                        }
+
+                        if(angular.isString(purchaseData.daily)){
+                            return purchaseData.daily === PURCHASED_ALL;
+                        }else{
+                            var maxAvailDailyOrder = (purchaseData.daily || 0) + (freeContent.daily || 0);
+                            return dailyOrder <= maxAvailDailyOrder;
+                        }
+                    });
+                }
 
-                    var purchaseData = res[0];
-                    var freeContent = res[1];
+                function isExamAvail(examId){
+                    return _baseIsEntityAvail().then(function(res){
+                        if(res === true){
+                            return true;
+                        }
 
-                    var examKeyProp = idToKeyInStorage(examId);
-                    var sectionKeyProp = idToKeyInStorage(sectionId);
+                        var purchaseData = res[0];
+                        var freeContent = res[1];
+                        var specials = res[2];
 
-                    var isExamPurchased = _isExamPurchased(purchaseData,examId);
-                    if(isExamPurchased ){
-                        return true;
-                    }
+                        var isPurchased = _isExamPurchased(purchaseData,examId);
```

Comment:
I guess this function can be deleted and _isFreeContent can be used instead(it should be renamed to isItemOwned)